api/src/main/java/io/minio/messages/ObjectLockConfiguration.java

```diff
@@ -45,7 +45,7 @@ public ObjectLockConfiguration(RetentionMode mode, RetentionDuration duration)
       this.rule = new Rule(mode, duration);
     }
 
-    if (mode != null || duration != null) {
+    if (mode == null || duration == null) {
```

Comment:
This fix is incorrect. Either both values non-null or null. i.e. `mode != null && duration != null` and `mode == null && duration == null` are valid.

Basically you need to have current `if` as `else if`